packages/react-form-renderer/src/form-renderer/enhanced-on-change.js

```diff
@@ -40,19 +41,45 @@ const convertType = (dataType, value) => ({
   [dataTypes.BOOLEAN]: castToBoolean(value),
 })[dataType] || value;
 
+/**
+ * Checks the value and returns undefined if its empty. Converst epmtry strings, arrays and objects.
+ * If value is empty its overriden to undefined for further processing.
+ * @param {Any} value Any JS variable to be check if is empty
+ */
+const checkEmpty = value => {
+  if (typeof value === 'number') {
+    return value;
+  }
+
+  if (typeof value === 'string' && value.length > 0) {
+    return value;
+  }
+
+  if (!isEmpty(value)) {
+    return value;
+  }
+
+  return undefined;
+};
+
 /**
  * Casts input value into selected data type
  * @param {FieldDataTypes} dataType intended data type of output value
  * @param {Function} onChange original function to be modified
+ * @param {Any} initial initial value of field
  * @param {Any} value value to be type casted
  * @param  {...any} args rest of orininal function arguments
  */
-const enhancedOnChange = (dataType, onChange, value, ...args) => {
+const enhancedOnChange = (dataType, onChange, initial, deletedValue, value, ...args) => {
   const sanitizedValue = sanitizeValue(value);
-  return onChange(
-    Array.isArray(sanitizedValue)
-      ? sanitizedValue.map(item => convertType(dataType, sanitizeValue(item)))
-      : convertType(dataType, sanitizedValue),
-    ...args);};
+  const result = Array.isArray(sanitizedValue)
+    ? sanitizedValue.map(item => convertType(dataType, sanitizeValue(item)))
+    : convertType(dataType, sanitizedValue);
+  if (typeof checkEmpty(result) === 'undefined' && typeof initial !== 'undefined') {
+    return onChange(deletedValue, ...args);
```

Comment:
Actually this may be problematic...

even if we don't use `deletedValue`, this will now always convert "" to `undefined`, right?

I think `""` and `undefined` are **very** different, unless the fields has a very specific meaning.

Reply:
`"" => undefined` is being done anyway by the form API. You can specify `parse` prop in field to change it. So this was happening even without the deleted value feature.